src/Shared/OpportunisticIntern.cs

```diff
@@ -465,6 +465,71 @@ internal StringInternTarget(string target)
             public bool ReferenceEquals(string other) => ReferenceEquals(_target, other);
         }
 
+        /// <summary>
+        /// Wrapper over a substring of a string.
+        /// </summary>
+        internal struct SubstringInternTarget : IInternable
```

Comment:
Could be `readonly struct`.

Reply:
Thank you, I'll make it `readonly` in a future PR.